### PR TITLE
@Alex -- Please review Mentoring Program 1St Session task

### PR DESCRIPTION
**Pull request by krushnadattaswain**

### CODE REVIEW

File: src/main/java/com/epam/processor/DataProcessor.java [hunk 1]
```diff
@@ -51,7 +77,18 @@ public RoadAccident getAccidentByIndex7(String index){
      * @return
      */
     public Map<String, Long> getCountByRoadSurfaceCondition7(){
-        return null;
+        
+    	Map<String, Long>  result = new HashMap<String, Long>();
+    	Multimap<String, RoadAccident> intermediateMap = (Multimap) ArrayListMultimap.create();
+     	for(RoadAccident accident : roadAccidentList) {
```
Comment (picaro) [inline src/main/java/com/epam/processor/DataProcessor.java:83]: You could do it in a one cycle - just increasing values inside one resulted map

File: src/main/java/com/epam/processor/DataProcessor.java [hunk 2]
```diff
@@ -71,18 +108,25 @@ public RoadAccident getAccidentByIndex7(String index){
...
 
-
-    public RoadAccident getAccidentByIndex(String index){
-        return null;
-    }
-
+	public RoadAccident getAccidentByIndex(String index) {
+		return roadAccidentList.stream()
+				.filter(accident -> accident.getAccidentId().equals(index))
+				.findFirst().orElse(null);
```
Comment (picaro) [inline src/main/java/com/epam/processor/DataProcessor.java:128]: Without Else you will have - NoSuchElementException  and it's even better, than NULL

File: src/main/java/com/epam/processor/DataProcessor.java [hunk 3]
```diff
@@ -92,32 +136,64 @@ public RoadAccident getAccidentByIndex(String index){
...
      */
-    public Collection<RoadAccident> getAccidentsByLocation(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude){
-        return null;
-    }
+	public Collection<RoadAccident> getAccidentsByLocation(float minLongitude,
+			float maxLongitude, float minLatitude, float maxLatitude) {
+		
+		return roadAccidentList
+				.stream()
+				.filter(roadAccident -> filterLocationbyCordinates(
```
Comment (picaro) [inline src/main/java/com/epam/processor/DataProcessor.java:144]: your method didn't do this string shorter :)    but it's up to you

File: src/main/java/com/epam/processor/DataProcessor.java [hunk 4]
```diff
@@ -92,32 +136,64 @@ public RoadAccident getAccidentByIndex(String index){
...
      * @return
      */
-    public List<String> getTopThreeWeatherCondition(){
-        return null;
-    }
+	public List<String> getTopThreeWeatherCondition() {
+		Map<String, Long> noOfIncidents = roadAccidentList.stream().collect(
+				groupingBy(RoadAccident::getWeatherConditions, counting()));
+
+		List<String> accidentListByWeatherCond = noOfIncidents
```
Comment (picaro) [inline src/main/java/com/epam/processor/DataProcessor.java:168]: Solution in a one string, Bravo!  Will check if its working :)

**Comment by picaro** [comment]
Regarding - getTopThreeWeatherCondition7
Don't try to do it in a beautiful way. You could do in a several steps
1. Fill Map  with a data NumEntries -> Weather 
2. Sort Map with Collections to the list of Map.Entries
3. Get first 3 elements

but that only 1 variant, you could do it as you want

File: src/main/java/com/epam/processor/DataProcessor.java (hunk 2 above)
Comment (bug-hunterx) [inline src/main/java/com/epam/processor/DataProcessor.java:128]: The contract declared by tests was to return null ;)

**Comment by bug-hunterx** [comment]
@krushnadattaswain @picaro Are you still working on this PR? I see Krushna's score is 90%. So just a bit left to get full score :) 

Krushna, if you are not going to finish it, please close PR

**Comment by bug-hunterx** [comment]
closing PR

**Pull request closed without merging**